### DISCUSSION TITLE
Can cancellation of the upload through progress callback be never called?

### DISCUSSION THREAD

**Discussion by ribtoks** [curl/curl#7388] [Q&A] [answered]
Hi

Consider a situation where we upload a file via FTP, but due to the network timeout, there's no progress for considerable amount of time (and TCP does not abort the connection for some time). We would like to cancel the upload, but upload progress never gets called since there's no "progress" per se.

Is it a possible situation? If yes, can we force curl to call "upload progress function" even with zero progress?

Thanks!

**Comment by jay** [comment, marked as answer]
The progress callback should be called by libcurl at least once a second except in some rare cases in the advanced event interface, or there is some function that was called and it blocks for more than a second. We try hard to avoid functions that are known as blocking, you can read more about it in [TODO 2.1 More non-blocking](https://curl.se/docs/todo.html#More_non_blocking)

**Reply by ribtoks** [reply]
Thanks, Jay
